File: src/main/java/itmo/is/project/model/module/dock/DockModule.java
package itmo.is.project.model.module.dock;

import itmo.is.project.model.module.Module;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "dock_module")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DockModule extends Module<DockModuleBlueprint> {


    @OneToMany(mappedBy = "dockModule", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<DockingSpot> dockingSpots;
}

File: src/main/kotlin/itmo/isproject/model/module/dock/DockModule.kt
package itmo.isproject.model.module.dock

import itmo.isproject.model.module.Module
import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "dock_module")
class DockModule(

    @OneToMany(mappedBy = "dockModule", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
    var dockingSpots: MutableList<DockingSpot> = ArrayList(),

    blueprint: DockModuleBlueprint? = null

) : Module<DockModuleBlueprint>(blueprint)
